packages/jaeger-ui/package.json

```diff
@@ -17,7 +17,7 @@
     "@babel/preset-typescript": "^7.21.0",
     "@testing-library/jest-dom": "^6.0.0",
     "@testing-library/react": "^14.0.0",
-    "@types/react": "16.14.35",
+    "@types/react": "18.2.22",
```

Comment:
@anshgoyalevil I thought we were blocked from doing this upgrade for some reason, but the CI seems green, so is this OK to merge? Or do we have an override elsewhere back to 16x?

Reply:
No no. The CI is green because we have a dependency resolution added for @types/react which states that "for any version of @types/react being required by a dependency, the yarn would fetch only the v16.

So, this PR is basically like upgrading types/react v18 in `package.json` but still fetching v16 from the node source

Reply:
Why do we need this extra dependency resolution setting instead of only keeping v16 in `package.json`? It makes it more difficult to understand what's happening.

Reply:
No. Let me explain:

- We have a resolution in `root_folder/package.json` file to fix the `@types/react` version to `16.14.35`

It means if there is a package that requires the `@types/react` v18, we would provide it with v16.14.35 forcefully.

For example, 
- let us suppose `ant-design` package is published to `npm registry` using `@types/react` `v18`. In other words, the development team of `ant-design` uses `@types/react` v18.
- So, whenever we would install `ant-design` package to out codebase, it would fetch `types/react` v18.
- But, what if out code is not ready for `v18`?
- `npm resolution` makes it easy by giving us the option to restrict the requirement of `ant-design` to `@types/react` v16.14.35

Now what this PR is trying to do is, it is treating `@types/react` as a package which is trying to fetch `@types/react` v18 from npm registry, but since we have a resolution added for it, it still provides the `@types/react` v16 to the codebase.

Reply:
Another example:

- PackageA on upgrade to vA wants @types/react v18
- PackageB on upgrade to vB wants @types/react v17
- But our codebase is only ready for @types/react v16

- Here comes npm resolutions

It says "I would provide v16 to PackageA and PackageB, no matter what they want."

Now, in package.json,
- PackageA gets listed as vA in the package.json, but internally it is provided with @types/react v16 only.
- Similarly, PackageB gets listed as vB in the package.json, but internally it is provided with @types/react v16 only.

Now, replace PackageB with @types/react in the whole scenario. That's what is happening here.